packages/voilite/index.js

```diff
@@ -0,0 +1,220 @@
+  // Handle commonjs or es2015 modules
+  let exports;
+  if (Object.prototype.hasOwnProperty.call(extension, '__esModule')) {
+    exports = extension.default;
+  } else {
+    // CommonJS exports.
+    exports = extension;
+  }
+
+  let plugins = Array.isArray(exports) ? exports : [exports];
+  for (let plugin of plugins) {
+    if (
+      PageConfig.Extension.isDisabled(plugin.id) ||
+      disabled.includes(plugin.id) ||
+      disabled.includes(plugin.id.split(':')[0])
+    ) {
+      continue;
+    }
+    yield plugin;
+  }
+}
+
+function loadScript(url) {
+  return new Promise((resolve, reject) => {
+    const newScript = document.createElement('script');
+    newScript.onerror = reject;
+    newScript.onload = resolve;
+    newScript.async = true;
+    document.head.appendChild(newScript);
+    newScript.src = url;
+  });
+}
+
+async function loadComponent(url, scope) {
+  await loadScript(url);
+
+  // From MIT-licensed https://github.com/module-federation/module-federation-examples/blob/af043acd6be1718ee195b2511adf6011fba4233c/advanced-api/dynamic-remotes/app1/src/App.js#L6-L12
+  // eslint-disable-next-line no-undef
+  await __webpack_init_sharing__('default');
+  const container = window._JUPYTERLAB[scope];
+  // Initialize the container, it may provide shared modules and may need ours
+  // eslint-disable-next-line no-undef
+  await container.init(__webpack_share_scopes__.default);
+}
+
+async function createModule(scope, module) {
+  try {
+    const factory = await window._JUPYTERLAB[scope].get(module);
+    return factory();
+  } catch (e) {
+    console.warn(
+      `Failed to create module: package: ${scope}; module: ${module}`
+    );
+    throw e;
+  }
+}
+
+const disabled = ['@jupyter-widgets/jupyterlab-manager'];
+
+/**
+ * The main function
+ */
+async function main() {
+  const litePluginsToRegister = [];
+
+  // Add the base serverlite extensions
+  // TODO Load these at the same time as federatedMimeExtensionPromises?
+  const baseServerExtensions = await Promise.all(serverExtensions);
+  baseServerExtensions.forEach((p) => {
+    for (let plugin of activePlugins(p)) {
+      litePluginsToRegister.push(plugin);
+    }
+  });
+
+  // create the in-browser JupyterLite Server
+  const jupyterLiteServer = new JupyterLiteServer({});
+  jupyterLiteServer.registerPluginModules(litePluginsToRegister);
+  // start the server
+  jupyterLiteServer.start();
+
+  const { serviceManager } = jupyterLiteServer;
+  await serviceManager.ready;
+
+  let mods = [
+    // @jupyterlab plugins
+    require('@jupyterlab/markdownviewer-extension'),
+    require('@jupyterlab/mathjax2-extension'),
+    require('@jupyterlab/rendermime-extension'),
+    // TODO: add the settings endpoint to re-enable the theme plugins?
+    // This would also need the theme manager plugin and settings
+    // require('@jupyterlab/theme-light-extension'),
+    // require('@jupyterlab/theme-dark-extension'),
+    require('./plugins')
```

Comment:
These are pretty much copy-pastes of the main Voila app, we should probably factorize it